### PR TITLE
SPR-16831 Allow null value for second attribute for ConcurrentModel.a…

### PR DESCRIPTION
**Pull request by sunriize**
…ddAttribute(String, Object) according to the documentation

### CODE REVIEW

File: spring-context/src/main/java/org/springframework/ui/ConcurrentModel.java [hunk 1]
```diff
@@ -16,14 +16,14 @@
...
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-
 import org.springframework.core.Conventions;
 import org.springframework.lang.Nullable;
 import org.springframework.util.Assert;
 
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
```
Comment (snicoll) [inline spring-context/src/main/java/org/springframework/ui/ConcurrentModel.java]: Please refrain from making formatting changes like these.

**Review by snicoll — COMMENTED** [review]
Thanks for the PR! This assertion is indeed out of sync with the `Nullable` annotation. I've added a comment, can you please take a look?

File: spring-context/src/main/java/org/springframework/ui/ConcurrentModel.java (hunk 1 above)
Comment (ddelamotte) [reply spring-context/src/main/java/org/springframework/ui/ConcurrentModel.java]: Sure, sorry i'll change it right now

**Comment by ddelamotte** [comment]
@snicoll I removed the extra file formating

**Comment by jhoeller** [comment]
The assertion is actually right; it's just the javadoc that is wrong. Overriding the `addAttribute` method from `Model` means that we formally have to declare it as `@Nullable` while this particular implementation doesn't actually support `null` values (it would end up as an NPE in `ConcurrentHashMap.put`).

**Pull request closed without merging**